stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -890,3 +890,38 @@ end
 *(A::Adjoint{<:Any,<:RealHermSymComplexHerm}, B::Transpose{<:Any,<:AbstractTriangular}) = A.parent * B
 *(A::Transpose{<:Any,<:RealHermSymComplexSym}, B::Adjoint{<:Any,<:AbstractTriangular}) = A.parent * B
 *(A::Transpose{<:Any,<:RealHermSymComplexSym}, B::Transpose{<:Any,<:AbstractTriangular}) = A.parent * B
+
+"""
+    symmetrize!(X::AbstractMatrix{<:AbstractFloat})
+
+Make the matrix `X` symmetric in-place using the formula `(X + X') / 2`.
+
+See also: [`symmetrize`](@ref)
+
+!!! compat "Julia 1.3"
+    This function requires Julia 1.3 or later.
+"""
+function symmetrize!(X::AbstractMatrix{<:AbstractFloat})
+    require_one_based_indexing(X)
```

Comment:
I think it wouldn't be hard to support arbitrary indexing. Wouldn't something like
```julia
idx = axes(X, 1)
jdx = axes(X, 2)
idx == jdx || error
for _j, _i...
  i, j = idx[_i], jdx[j]
 ...
end
```
work? Hopefully, the compiler is able to do something useful when the input has `OneTo` indices.

Reply:
I was lazy and had already implemented the 1-indexed version so I didn't bother trying with others. :stuck_out_tongue: I'll give this a try.

Reply:
I've implemented it such that non-one-based indexing is supported, but the axes need to have the same indexing.